core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2052,6 +2051,31 @@ private[spark] object Utils extends Logging {
     }
   }
 
+
+  private[this] val nonSpaceOrNaturalLineDelimiter: Char => Boolean = {
```

Comment:
This can be declared inside `trimExceptCRLF`.

Also, nit: `ch =>` stays with the opening brace.